[Doc] Add GRU new gate calculation difference

isdanni opened the pull request:
Summary: Add a note for the calculation difference of GRU new gate `n_t` between PyTorch and original paper.

Fix: #99531 

Test Plan: Please see GitHub pipelines.

Differential Revision: D45579790

mikaylagawarecki commented on torch/ao/nn/quantized/dynamic/modules/rnn.py:
```diff
@@ -610,6 +610,26 @@ class GRU(RNNBase):
         where :math:`k = \frac{1}{\text{hidden\_size}}`
 
+    .. note::
+        The calculation of new gate :math:`n_t` subtly differs from the orignal paper and other frameworks.
```
nit, and same below in torch.nn.GRU
```suggestion
        The calculation of new gate :math:`n_t` subtly differs from the original paper and other frameworks.
```

mikaylagawarecki reviewed (approved):
Looks great, thanks! Just one typo

isdanni committed:
[Doc] Add GRU new gate calculation difference (#100646)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/100646

Add a note for the calculation difference of GRU new gate `n_t` between PyTorch and original paper.

Test Plan: Please see GitHub pipelines.

Differential Revision: D45579790

fbshipit-source-id: 02a17ebe6d3c7dfb65ff9b8265a79bcd059aecec

mikaylagawarecki commented:
@pytorchbot merge
